r/tests/testthat/test-Array.R

```diff
@@ -723,6 +723,17 @@ test_that("[ accepts Arrays and otherwise handles bad input", {
   )
 })
 
+test_that("[ %in% looks up string key in dictionary", {
+  a1 <- Array$create(as.factor(c("A", "B", "C")))
+  a2 <- DictionaryArray$create(c(0L, 1L, 2L), c(4.5, 3.2, 1.1))
+  b1 <- Array$create(c(FALSE, TRUE, FALSE))
+  b2 <- Array$create(c(FALSE, FALSE, FALSE))
+  expect_equal(b1, arrow:::call_function("is_in_meta_binary", a1, Array$create("B")))
+  expect_equal(b2, arrow:::call_function("is_in_meta_binary", a1, Array$create("D")))
+  expect_equal(b1, arrow:::call_function("is_in_meta_binary", a2, Array$create(3.2)))
+  expect_error(arrow:::call_function("is_in_meta_binary", a2, Array$create("B")))
```

Comment:
I would write these as e.g. `expect_equal(a1 %in% "B", b1)` etc.

Reply:
@jonkeane reminds me that `%in%` isn't an option here due to R's method dispatch quirks. So what I suggested won't work.

In that case, keep these as they are, but (1) drop the `arrow:::` (not needed in the test suite, it evals in the package namespace) and (2) switch the order of arguments so that the expected result is second.

Reply:
I was just about to ask about this. @jonkeane told me about this quirk.
I've proposed a new [`is_in`](https://github.com/apache/arrow/pull/9606/files#diff-9d33f036186210a1076f3adc590e0cc60a0b65556cefdbd15e6d26aa105f7d76R120) function. If it doesn't make sense for consistency we can scrap it.
Is there a plan to create aliases for calling call_function at some point?

Reply:
Within the context of a dplyr method, we can do nonstandard evaluation and map anything we want to the appropriate arrow compute functions, and we do that. But when using R's standard evaluation, we can only define arrow methods where the R function is defined as a generic. We do this for (most) arithmetic and logical operations, for example. However, `%in%` is not defined that way (nor is `match()`, the lower-level function it calls), so we can't. 

I'm ok with adding an `is_in()` function, though I'm also ok with not doing it as I don't think it will get used much. What do you think @jonkeane @ianmcook ?

Reply:
I think adding `is_in()` is a good idea, especially since it'll give us a nice place to document this particular quirk

Reply:
It should be documented along with `match_arrow()`, which as it turns out mentions the same quirk https://github.com/apache/arrow/blob/master/r/R/compute.R#L88

Reply:
I added [some doc text](https://github.com/apache/arrow/pull/9606/files#diff-9d33f036186210a1076f3adc590e0cc60a0b65556cefdbd15e6d26aa105f7d76R112) to `is_in`. It's a bit of copy-pasta, please review.

Shall we also add something to the [r/NEWS.md](https://github.com/apache/arrow/blob/master/r/NEWS.md)?

Reply:
> Shall we also add something to the r/NEWS.md?

Yeah probably, but I'll come back to that before the release, we're a little backed up on changelog updates already.